x-pack/plugins/actions/server/builtin_action_types/teams.ts

```diff
@@ -0,0 +1,229 @@
+  };
+}
+
+function validateActionTypeConfig(
+  configurationUtilities: ActionsConfigurationUtilities,
+  secretsObject: ActionTypeSecretsType
+) {
+  let url: URL;
+  try {
+    url = new URL(secretsObject.webhookUrl);
+  } catch (err) {
+    return i18n.translate('xpack.actions.builtin.teams.teamsConfigurationErrorNoHostname', {
+      defaultMessage: 'error configuring teams action: unable to parse host name from webhookUrl',
+    });
+  }
+
+  try {
+    configurationUtilities.ensureHostnameAllowed(url.hostname);
+  } catch (allowListError) {
+    return i18n.translate('xpack.actions.builtin.teams.teamsConfigurationError', {
+      defaultMessage: 'error configuring teams action: {message}',
+      values: {
+        message: allowListError.message,
+      },
+    });
+  }
+}
+
+// action executor
+
+async function teamsExecutor(
+  { logger }: { logger: Logger },
+  execOptions: TeamsActionTypeExecutorOptions
+): Promise<ActionTypeExecutorResult<unknown>> {
+  const actionId = execOptions.actionId;
+  const secrets = execOptions.secrets;
+  const params = execOptions.params;
+  const { webhookUrl } = secrets;
+  const { message } = params;
+  const data = { text: message };
+
+  const axiosInstance = axios.create();
+
+  const result: Result<AxiosResponse, AxiosError> = await promiseResult(
+    request({
+      axios: axiosInstance,
+      method: 'post',
+      url: webhookUrl,
+      logger,
+      data,
+      proxySettings: execOptions.proxySettings,
+    })
+  );
+
+  if (isOk(result)) {
+    const {
+      value: { status, statusText, data: responseData, headers: responseHeaders },
+    } = result;
+
+    // Microsoft Teams connectors do not throw 429s. Rather they will return a 200 response
+    // with a 429 message in the response body when the rate limit is hit
+    // https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using#rate-limiting-for-connectors
+    if (isString(responseData) && responseData.includes('ErrorCode:ApplicationThrottled')) {
+      return pipe(
+        getRetryAfterIntervalFromHeaders(responseHeaders),
+        map((retry) => retryResultSeconds(actionId, message, retry)),
+        getOrElse(() => retryResult(actionId, message))
+      );
+    }
+
+    logger.debug(`response from teams action "${actionId}": [HTTP ${status}] ${statusText}`);
+
+    return successResult(actionId, data);
+  } else {
+    const { error } = result;
+
+    if (error.response) {
+      const { status, statusText } = error.response;
+      const serviceMessage = `[${status}] ${statusText}`;
+      logger.error(`error on ${actionId} Microsoft Teams event: ${serviceMessage}`);
+
+      // special handling for 5xx
+      if (status >= 500) {
+        return retryResult(actionId, serviceMessage);
+      }
+
+      return errorResultInvalid(actionId, serviceMessage);
+    }
+
+    logger.error(`error on ${actionId} Microsoft Teams action: unexpected error`);
```

Comment:
This might be superfluous now, we added some action error logging in 7.9 or 7.10.  I wonder if it was just for action executions with alerts though ... looking ... we seem to log in the action executor and not just for alerts: https://github.com/elastic/kibana/blob/fed9a4fddcc0087ee9eca6582a2a84e001890f08/x-pack/plugins/actions/server/lib/action_executor.ts#L167-L184

Changing these to a `debug` is not a bad idea, vs removing it, and I guess double-check that the message you see in the log matches the same info that these log messages produce.